formula-android/src/main/java/com/instacart/formula/fragment/FragmentFlowStore.kt

```diff
@@ -87,12 +87,14 @@ class FragmentFlowStore(
                         is LifecycleEvent.Removed -> {
                             val updated = state.copy(
                                 activeKeys = state.activeKeys.minus(key),
-                                states = state.states.minus(key)
+                                states = state.states.minus(key),
+                                features = state.features.minus(key)
                             )
                             transition(updated)
                         }
                         is LifecycleEvent.Added -> {
                             if (!state.activeKeys.contains(key)) {
+                                // TODO: should use input.onScreenError() instead of adding fake render model!
```

Comment:
Still todo?

Reply:
Yeah, I noticed this improvement while working on this feature, but will actually implement it in another PR.